### PR TITLE
Added metadata examples for Video, Recipe, and Review types

### LINKED ISSUE #491: Build samples for other static content types
e.g. Blog, Recipe, TechArticle

### PR DESCRIPTION
**Pull request by nzukoski**
Also moves Article metadata into the same folder.

This pull request fixes #491

### CODE REVIEW

File: examples/metadata-examples/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+<pre>
+  article-json-ld.amp.html       - JSON-LD structured data markup format for Articles
+  article-microdata.amp.html     - microdata structured data markup format for Articles
+  recipe-json-ld.amp.html       - JSON-LD structured data markup format for Recipes
+  recipe-microdata.amp.html     - microdata structured data markup format for Recipes
+  review-json-ld.amp.html       - JSON-LD structured data markup format for Reviews
+  review-microdata.amp.html     - microdata structured data markup format for Reviews
+  video-json-ld.amp.html       - JSON-LD structured data markup format for Videos
+  video-microdata.amp.html     - microdata structured data markup format for Videos
```
Comment (jmadler) [inline examples/metadata-examples/README.md]: Can you vertically align the notes?

File: examples/metadata-examples/video-json-ld.amp.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    </script>
+    <!-- this style tag is required -->
+    <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+  </head>
+  <body>
+    <h2>The Audience is programing</h2>
+    <p>A programming audience</p>
+    <amp-video
```
Comment (jmadler) [inline examples/metadata-examples/video-json-ld.amp.html]: The src URL must be absolute (and https or schema non specific)

File: examples/metadata-examples/video-json-ld.amp.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        //
+      {
+        "@context": "http://schema.org",
+        "@type": "VideoObject",
+        "name": "The Audience is programing",
+        "description": "A programming audience",
+        "thumbnailUrl": "http://cdn.ampproject.org/leader.jpg",
+        "uploadDate": "2015-02-05T08:00:00+08:00",
+        "duration": "PT1M33S",
+        "contentUrl": "http://cdn.ampproject.org/examples/video/The-Audience-Is-Programming.mp4",
```
Comment (jmadler) [inline examples/metadata-examples/video-json-ld.amp.html]: Use HTTPS please

Comment (jmadler) [inline examples/metadata-examples/video-json-ld.amp.html]: (just for the contentUrl, to match the below)

File: examples/metadata-examples/video-microdata.amp.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<!doctype html>
```
Comment (jmadler) [inline examples/metadata-examples/video-microdata.amp.html]: Same comments as ld-json version

Comment (nzukoski) [inline examples/metadata-examples/video-microdata.amp.html]: Done.

File: examples/metadata-examples/video-json-ld.amp.html (hunk 2 above)
Comment (nzukoski) [inline examples/metadata-examples/video-json-ld.amp.html]: Done.

**Comment by jmadler** [comment]
LGTM - please squash and also handle CLA

**Comment by nzukoski** [comment]
I signed the CLA

**Comment by nzukoski** [comment]
Took me a little while to figure out the squashing but it should be good to go now.

**Comment by jmadler** [comment]
Update example filenames in these two places:

https://github.com/ampproject/amphtml/blob/master/gulpfile.js#L298
https://github.com/ampproject/amphtml/blob/master/test/integration/test-example-validation.js#L37

**Comment by nzukoski** [comment]
Fixed the example file names and also updated the mainEntityOfPage stuff for the json-ld examples. Not sure why the Travis CI build is still failing...

**Comment by erwinmombay** [comment]
@nzukoski was just a flakey test. so the travis ci build failing isnt this PR's fault. do you might resubmitting to make it run again? thanks!

**Comment by nzukoski** [comment]
All tests passing now.

**Comment by jmadler** [comment]
LGTM

**Pull request merged**